check_yum.py

```diff
@@ -322,33 +322,25 @@ def get_all_updates(self):
         if self.verbosity >= 4:
             for section in output2:
                 print("\nSection:\n%s\n" % section)
-        if len(output2) > 2 or \
+        if len(output2) > 2 or (not self.no_cache_update and \
            not ("Setting up repositories" in output2[0] or \
                 "Loaded plugins: " in output2[0] or \
                 "Last metadata expiration check" in output2[0] or \
-                re.search(r'Loading\s+".+"\s+plugin', output2[0])):
+                re.search(r'Loading\s+".+"\s+plugin', output2[0]))):
             end(WARNING, "Yum output signature does not match current known "  \
                        + "format. " + support_msg)
         num_packages = 0
-        if len(output2) == 1:
+        if len(output2) == 1 and not self.no_cache_update:
             # There are no updates but we have passed
             # the loading and setting up of repositories
             pass
         else:
-            for line in output2[1].split("\n"):
+            for line in output2[-1].split("\n"):
```

Comment:
Instead of taking the second element, we now take the last one. This works for the cache and non-cache part the same (we already checked above that there are not more than two entries).